core/src/Kokkos_Complex.hpp

```diff
@@ -829,4 +829,13 @@ struct reduction_identity<Kokkos::complex<T> > {
 
 } // namespace Kokkos
 
+namespace std
+{
+  template <typename Number>
+  Number abs( Kokkos::complex<Number> n )
+  {
+    return std::sqrt(n.real()*n.real()+n.imag()*n.imag());
+  }
+}
```

Comment:
I don't think it is right to inject that overload in the `std` namespace and I don't see why you would need to do so.  It would be found it by ADL if you put it in the `Kokkos` namespace and make an unqualified call to `abs()`.

Also note that `std::abs()` take the complex number by const reference.

Reply:
Another thought: should it be a `KOKKOS_INLINE_FUNCTION`?

Reply:
Yes, thanks. There is already `Kokkos::abs` and an unqualified call should work.